microhapulator/data/template.html

```diff
@@ -1,234 +1,324 @@
+                <th class="alnrt">Retention</th>
+            </tr>
+            {% for sample, ambig_reads_df in ambiguous_reads.items() %}
+            <tr>
+                <td>{{sample}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["R1OnlyFailed"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["R2OnlyFailed"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["PairsFailed"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["PairsRemoved"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["PairsKept"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["TotalPairs"])}}</td>
+                <td class="alnrt">{{"{:.2f}".format(ambig_reads_df.iloc[0]["PairsKept"] /
+                    ambig_reads_df.iloc[0]["TotalPairs"] * 100)}}%</td>
+            </tr>
+            {% endfor %}
+        </table>
+        {% else %}
+        <p>
+            If over {{"{:.0f}".format(ambiguous_read_threshold * 100)}}% of bases in a read are ambiguous ('N'), it is
+            filtered out prior to read mapping and excluded from all subsequent analysis.
+            The table below shows the number of reads removed from each sample.
+        </p>
+        <table>
+            <tr>
+                <th>Sample</th>
+                <th class="alnrt">Reads Removed</th>
+                <th class="alnrt">Reads Kept</th>
+                <th class="alnrt">Total Reads</th>
+                <th class="alnrt">Retention</th>
+            </tr>
+            {% for sample, ambig_reads_df in ambiguous_reads.items() %}
+            <tr>
+                <td>{{sample}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["ReadsRemoved"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["ReadsKept"])}}</td>
+                <td class="alnrt">{{"{:,}".format(ambig_reads_df.iloc[0]["ReadsRemoved"] +
+                    ambig_reads_df.iloc[0]["ReadsKept"])}}</td>
+                <td class="alnrt">{{"{:.2f}".format(ambig_reads_df.iloc[0]["ReadsKept"] /
+                    (ambig_reads_df.iloc[0]["ReadsRemoved"] + ambig_reads_df.iloc[0]["ReadsKept"]) * 100)}}%</td>
+            </tr>
+
+            {% endfor%}
+            {% endif %}
+        </table>
+        {% if not reads_are_paired %}
+        <p> Merged reads less than {{"{:,}".format(read_length_threshold)}} base pairs are filtered out prior to read
+            mapping and excluded from all subsequent analysis. The following table shows the number of merged reads
+            removed from each sample.
+        </p>
+        
+        <table>
+            <tr>
+                <th>Sample</th>
+                <th class="alnrt">Reads Removed</th>
+                <th class="alnrt">Reads Kept</th>
+                <th class="alnrt">Total Unambiguous Reads</th>
```

Comment:
I know we talked about this being a mouthful. But I don't want to use `Total Reads`... because its not the total number of reads. And `Filtered Reads` here seems too vague. Very open to ideas for better column header names